src/Analysis/Codelyzer.Analysis/CSharpCodeAnalyzer.cs

```diff
@@ -39,6 +40,35 @@ public override async Task<List<AnalyzerResult>> AnalyzeSolution(string solution
             return await Analyze(solutionPath);
         }
 
+        ///<inheritdoc/>
+        public override CodeGraph GenerateGraph(List<AnalyzerResult> analyzerResults)
+        {
+
+            var codeGraph = new CodeGraph();
+            try
+            {
+                codeGraph.Initialize(analyzerResults);
+            }
+            catch (Exception ex)
+            {
+                Logger.LogError(ex, "Error while generating graph");
+            }
+            return codeGraph;
+        }
+
+        ///<inheritdoc/>
+        public override async Task<SolutionAnalyzerResult> AnalyzeSolutionWithGraph(string solutionPath)
+        {
+            var analyzerResults = await AnalyzeSolution(solutionPath);
```

Comment:
we should convert the AnalyzeSolution to Value<Task>..  not part of this CR

Reply:
We can discuss separately as part of a new CR, as you mentioned

Reply:
please create an issue and add as comment => resolving this